validateWorkerVersions.ps1

```diff
@@ -0,0 +1,87 @@
+<#
+    .SYNOPSIS
+        Used to validate and/or update worker package versions
+    .EXAMPLE
+        ./validateWorkerVersions.ps1
+
+        Validates the workers match the existing host version and throws an error if they don't
+    .EXAMPLE
+        ./validateWorkerVersions.ps1 -Update -HostVersion 4.0.0
+
+        Updates the host reference to 4.0.0 and the workers to their matching versions
+#>
+param (
+    [Switch]$Update,
+    
+    # An explicit host version, otherwise the host version from Azure.Functions.Cli.csproj will be used
+    [string]$hostVersion
+)
+
+# the xml will fail to parse if the data is encoded with a bom character
+function removeBomIfExists([string]$data)
+{
+    if ($data.StartsWith(0xFEFF)) {
+        $data = $data.substring(1)
+    }
+    return $data
+}
+
+$cliCsprojPath = "./src/Azure.Functions.Cli/Azure.Functions.Cli.csproj"
+$cliCsprojContent = removeBomIfExists(Get-Content $cliCsprojPath)
+$cliCsprojXml = [xml]$cliCsprojContent
+
+function getPackageVersion([string]$packageName, [string]$csprojContent)
+{
+    $version = (Select-Xml -Content $csprojContent -XPath "/Project//PackageReference[@Include='$packageName']/@Version").ToString()
+    if (-Not $version) {
+        throw "Failed to find version for package $packageName"
+    }
+    return $version
+}
+
+function setCliPackageVersion([string]$packageName, [string]$version)
+{
+    $node = $cliCsprojXml.SelectSingleNode("/Project//PackageReference[@Include='$packageName']")
+    $node.Version = $version
+}
+
+$hostPackageName = "Microsoft.Azure.WebJobs.Script.WebHost"
+if (-Not $hostVersion) {
+    $hostVersion = getPackageVersion $hostPackageName $cliCsprojContent
+} elseif ($Update) {
+    setCliPackageVersion $hostPackageName $hostVersion
+}
+
+$hostRepoUrlBase = "https://raw.githubusercontent.com/Azure/azure-functions-host/v$hostVersion"
+$hostCsprojContent = removeBomIfExists((Invoke-WebRequest -Uri "$hostRepoUrlBase/src/WebJobs.Script/WebJobs.Script.csproj").Content)

```

Comment:
Consider adding `-MaximumRetryCount` `-RetryIntervalSec` to retry in case the first `Invoke-WebRequest` fails.